server/util/lockmap/lockmap_test.go

```diff
@@ -50,10 +56,16 @@ func TestRLock(t *testing.T) {
 	}
 	eg.Wait()
 	require.Equal(t, 20, m["samekey"])
+
+	// Wait for gc
+	time.Sleep(200 * time.Millisecond)
```

Comment:
Can we use more explicit waiting here? This seems like it will be flaky.

Reply:
I'm now waiting for the gc to clean up the element. This means that the test will time out if it fails to do so, but that seems better than either having it be flaky or depending on further `lockmap` internals.

Reply:
sounds good!